### ISSUE TITLE
Not building in Linux 6.12.1

### ISSUE THREAD

**Issue by Smith1970** [pop-os/system76-dkms#70]
**Distribution (run `cat /etc/os-release`):** Arch Linux
 
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):** gcc 14.2.1+r134+gab884fffe3fc-1

**Issue/Bug Description:** Not building in Linux 6.12.1

**Steps to reproduce (if you know):** 
make

**Expected behavior:** 
To build the module properly.

**Other Notes:**
```
make: Entering directory '/usr/lib/modules/6.12.1-zen1-1-zen/build'
  CC [M]  /var/lib/dkms/system76/1.0.16/build/system76.o
In file included from /var/lib/dkms/system76/1.0.16/build/system76.c:110:
/var/lib/dkms/system76/1.0.16/build/system76_hwmon.c:24:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
   24 | #if S76_HAS_HWMON
      |     ^~~~~~~~~~~~~
/var/lib/dkms/system76/1.0.16/build/system76_hwmon.c:24:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
/var/lib/dkms/system76/1.0.16/build/system76_hwmon.c:24:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
/var/lib/dkms/system76/1.0.16/build/system76.c: In function 's76_probe':
/var/lib/dkms/system76/1.0.16/build/system76.c:210:58: error: passing argument 2 of 'wmi_install_notify_handler' from incompatible pointer type [-Wincompatible-pointer-types]
  210 |         err = wmi_install_notify_handler(S76_EVENT_GUID, s76_wmi_notify, NULL);
      |                                                          ^~~~~~~~~~~~~~
      |                                                          |
      |                                                          void (*)(u32,  void *) {aka void (*)(unsigned int,  void *)}
In file included from /var/lib/dkms/system76/1.0.16/build/system76.c:25:
./include/linux/acpi.h:404:60: note: expected 'wmi_notify_handler' {aka 'void (*)(union acpi_object *, void *)'} but argument is of type 'void (*)(u32,  void *)' {aka 'void (*)(unsigned int,  void *)'}
  404 |                                         wmi_notify_handler handler, void *data);
      |                                         ~~~~~~~~~~~~~~~~~~~^~~~~~~
make[2]: *** [scripts/Makefile.build:229: /var/lib/dkms/system76/1.0.16/build/system76.o] Error 1
make[1]: *** [/usr/lib/modules/6.12.1-zen1-1-zen/build/Makefile:1934: /var/lib/dkms/system76/1.0.16/build] Error 2
make: *** [Makefile:224: __sub-make] Error 2
make: Leaving directory '/usr/lib/modules/6.12.1-zen1-1-zen/build'
```

**Comment by LorenAmelang** [comment]
I'm seeing a similar list of errors in my latest Arch update:

```
DKMS make.log for system76/1.0.13 for kernel 6.12.9-arch1-1 (x86_64)
Wed Jan 15 08:15:28 PM PST 2025
make: Entering directory '/usr/lib/modules/6.12.9-arch1-1/build'
  CC [M]  /var/lib/dkms/system76/1.0.13/build/system76.o
In file included from /var/lib/dkms/system76/1.0.13/build/system76.c:110:
...
```

Maybe mine is a different problem? Or this fix doesn't propagate beyond Pop-OS? 

Sorry if I'm an ignorant newbie to this level of Linux!

[make.log](https://github.com/user-attachments/files/18466538/make.log)

**Comment by t3hmrman** [comment]
Same, running into the same problem here on an `Oryx Pro orpy5` w/ Arch, on kernel 6.13.3

```
❯ cat /var/lib/dkms/system76/1.0.9/build/make.log 
DKMS (dkms-3.1.5) make.log for system76/1.0.9 for kernel 6.13.3-arch1-1 (x86_64)
Fri Feb 21 03:28:31 PM JST 2025
Cleaning build area
# command: make -C /usr/lib/modules/6.13.3-arch1-1/build M=/var/lib/dkms/system76/1.0.9/build clean
make: Entering directory '/usr/lib/modules/6.13.3-arch1-1/build'
make[1]: Entering directory '/var/lib/dkms/system76/1.0.9/build'
make[1]: Leaving directory '/var/lib/dkms/system76/1.0.9/build'
make: Leaving directory '/usr/lib/modules/6.13.3-arch1-1/build'

# exit code: 0
# elapsed time: 00:00:00
Building module(s)
# command: make -j12 KERNELRELEASE=6.13.3-arch1-1 -C /usr/lib/modules/6.13.3-arch1-1/build M=/var/lib/dkms/system76/1.0.9/build
make: Entering directory '/usr/lib/modules/6.13.3-arch1-1/build'
make[1]: Entering directory '/var/lib/dkms/system76/1.0.9/build'
  CC [M]  system76.o
In file included from system76.c:109:
system76_hwmon.c:24:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
   24 | #if S76_HAS_HWMON
      |     ^~~~~~~~~~~~~
system76_hwmon.c:24:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
system76_hwmon.c:24:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
system76.c: In function ‘s76_probe’:
system76.c:208:58: error: passing argument 2 of ‘wmi_install_notify_handler’ from incompatible pointer type [-Wincompatible-pointer-types]
  208 |         err = wmi_install_notify_handler(S76_EVENT_GUID, s76_wmi_notify, NULL);
      |                                                          ^~~~~~~~~~~~~~
      |                                                          |
      |                                                          void (*)(u32,  void *) {aka void (*)(unsigned int,  void *)}
In file included from system76.c:25:
/usr/lib/modules/6.13.3-arch1-1/build/include/linux/acpi.h:404:60: note: expected ‘wmi_notify_handler’ {aka ‘void (*)(union acpi_object *, void *)’} but argument is of type ‘void (*)(u32,  void *)’ {aka ‘void (*)(unsigned int,  void *)’}
  404 |                                         wmi_notify_handler handler, void *data);
      |                                         ~~~~~~~~~~~~~~~~~~~^~~~~~~
system76.c: At top level:
system76.c:281:19: error: initialization of ‘void (*)(struct platform_device *)’ from incompatible pointer type ‘int (*)(struct platform_device *)’ [-Wincompatible-pointer-types]
  281 |         .remove = s76_remove,
      |                   ^~~~~~~~~~
system76.c:281:19: note: (near initialization for ‘s76_platform_driver.remove’)
make[3]: *** [/usr/lib/modules/6.13.3-arch1-1/build/scripts/Makefile.build:196: system76.o] Error 1
make[2]: *** [/usr/lib/modules/6.13.3-arch1-1/build/Makefile:1989: .] Error 2
make[1]: *** [/usr/lib/modules/6.13.3-arch1-1/build/Makefile:251: __sub-make] Error 2
make[1]: Leaving directory '/var/lib/dkms/system76/1.0.9/build'
make: *** [Makefile:251: __sub-make] Error 2
make: Leaving directory '/usr/lib/modules/6.13.3-arch1-1/build'
```
